orchestration/src/test/java/com/sap/ai/sdk/orchestration/OrchestrationUnitTest.java

```diff
@@ -339,26 +344,74 @@ private static MaskingModuleConfig createMaskingConfig(
   }
 
   @Test
-  void testGenericErrorHandling() {
-    stubFor(post(anyUrl()).willReturn(serverError()));
-
-    assertThatThrownBy(() -> client.executeRequest(mock(CompletionPostRequest.class)))
-        .isInstanceOf(OrchestrationClientException.class)
-        .hasMessageContaining("500 Server Error");
-  }
-
-  @Test
-  void testOrchestrationErrorParsing() {
+  void testErrorHandling() {
     stubFor(
         post(anyUrl())
+            .inScenario("Errors")
+            .whenScenarioStateIs(Scenario.STARTED)
+            .willReturn(serverError())
+            .willSetStateTo("1"));
+    stubFor(
+        post(anyUrl())
+            .inScenario("Errors")
+            .whenScenarioStateIs("1")
             .willReturn(
                 badRequest()
                     .withHeader("Content-Type", "application/json")
-                    .withBodyFile("errorResponse.json")));
+                    .withBodyFile("errorResponse.json"))
+            .willSetStateTo("2"));
+    stubFor(
+        post(anyUrl())
+            .inScenario("Errors")
+            .whenScenarioStateIs("2")
+            .willReturn(
```

Comment:
(Minor/Comment)

I would rather chain `.willReturn(..)` than using scenarios and state-id. I don't see any benefit for that. But maybe I'm missing something.

Reply:
Probably true, but since this is copy/pasted code I'd not change it here but after refactoring later..